src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,54 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimal() {
+    for (let i = 0; i < Animal.alive.length; i++) {
+      if (Animal.alive[i].health <= 0) {
+        Animal.alive.splice(i, 1);
+        i--;
+      }
+    }
+  }
+
+  changeHealth(amount) {
+    this.health += amount;
+
+    if (this.health <= 0) {
+      this.deleteAnimal();
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
```

Comment:
The `hide` method should only set the `hidden` property to `true` without toggling it. Ensure that this method does not change the state if the animal is already hidden.